hibike/lib/hibike/util.cpp

```diff
@@ -0,0 +1,20 @@
+#include "util.h"
+
+// Assumes little endian
+
+uint64_t read_num_bytes(uint8_t* data, size_t num) {
+  uint64_t value = 0;
+  for (size_t i = 0; i < num; i++) {
+    value += data[i] << (8 * i);
+  }
+  return value;
+}
+
+
+// Assumes little endian
+
+void write_num_bytes(uint64_t data, uint8_t* data_buf, size_t num) {
```

Comment:
This function should be content-agnostic. 
Regardless of what the type of data we're read/writing is, we *pretend* it is a `uint64_t`. As long as we are consistent in how we package / unpackage the data, and in what we are returning to the calling function, we can have the application deal with interpreting the data. 

Using a pointer to data is probably a better solution. I recommend you make that upgrade in a future commit. It will require a minor refactor of the rest of the code as well.